src/confd/src/ietf-interfaces.c

```diff
@@ -996,24 +991,17 @@ static bool netdag_must_del(struct lyd_node *dif, struct lyd_node *cif)
 {
 	const char *iftype = lydx_get_cattr(cif, "type");
 
-	if (!strcmp(iftype, "infix-if-type:bridge")) {
-		if (is_phys_addr_deleted(dif))
-			return true;
-	} else if (!strcmp(iftype, "infix-if-type:vlan")) {
+	if (strcmp(iftype, "infix-if-type:ethernet")) {
 		if (is_phys_addr_deleted(dif))
 			return true;
+	}
 
-		if (lydx_get_cattr(dif, "parent-interface") ||
-		    lydx_get_descendant(lyd_child(dif),
-					"encapsulation",
-					"dot1q-vlan",
-					"outer-tag",
-					NULL))
+	if (!strcmp(iftype, "infix-if-type:vlan")) {
+		if (lydx_get_descendant(lyd_child(dif),
+								"vlan",
+								NULL))
```

Comment:
Another odd indent, that also can be collapsed to a single line (the second `if()`).